### DISCUSSION TITLE
Does arguably support stringified annotations?

### DISCUSSION THREAD

**Discussion by pawamoy** [treykeown/arguably#21] [Q&A] [answered]
If I add `from __future__ import annotations` to my code, will arguably handle it just fine?

(btw search is not enabled in your docs)

**Comment by treykeown** [comment]
Hi! Just checked - most tests pass, but ones related to `@arguably.arg.builder` are failing - see #22. I'll see if it's a quick fix.

Thanks for calling out search! Still learning all the incantations for `mkdocs` 😅

**Reply by treykeown** [reply, marked as answer]
Looks like #22 was just an artifact of the types being defined within a pytest fixture. Running https://github.com/treykeown/arguably/blob/main/etc/scripts/build.py with `from __future__ import annotations` at the top works just fine.

To answer your question - yes, it's supported. The caveat is the [normal troubles](https://stackoverflow.com/questions/69032511/how-to-get-callable-class-objects-from-self-annotations-with-future-anno/69034361#69034361) you'll encounter with using `get_type_hints` with `from __future__ import annotations`. For `arguably`, classes defined in a local scope will not work for type hints.

**Reply by pawamoy** [reply]
OK great, thanks! Yep, agreed on the normal troubles, well exposed by Pydantic :slightly_smiling_face:
